sentry_sdk/db/explain_plan/__init__.py

```diff
@@ -17,35 +17,44 @@ def cache_statement(statement):
     EXPLAIN_CACHE[hash(statement)] = datetime.datetime.utcnow()
 
 
-def should_run_explain_plan(statement, options):
-    # type: (str, dict[str, Any]) -> bool
+def remove_exired_cache_items(options):
+    # type: (dict[str, Any]) -> None
     """
-    Check cache if the explain plan for the given statement should be run.
+    Remove expired cache items from the cache.
     """
-    explain_cache_size = options.get("explain_cache_size", EXPLAIN_CACHE_SIZE)
+    global EXPLAIN_CACHE
+
     explain_cache_timeout_seconds = options.get(
         "explain_cache_timeout_seconds", EXPLAIN_CACHE_TIMEOUT_SECONDS
     )
 
     now = datetime.datetime.utcnow()
-    key = hash(statement)
 
+    for key, statement_timestamp in EXPLAIN_CACHE.items():
+        expiration_time = statement_timestamp + datetime.timedelta(
+            seconds=explain_cache_timeout_seconds
+        )
```

Comment:
We could calculate and save the `expiration_time` in the cache directly instead of the time of the statement. Then we don't need to recalculate here.

Reply:
uh, good idea! optimize all the things! (because the cache is in memory that makes sense, because one can not change the timeout on the fly so the expiration time can not change during the life of the cache.) will do this.